avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+            process.system(unexport_cmd)
+        else:
+            logging.warn("Target %s %s is not exported yet."
+                         "Can not unexport it." % (self.client, self.path))
+
+    def reset_export(self):
+        """
+        Reset the exportfs to the original status before we export the
+        specific entry.
+        """
+        self.unexport()
+        if self.ori_options:
+            tmp_options = self.options
+            self.options = self.ori_options.split(",")
+            self.export()
+            self.options = tmp_options
+
+    def export(self):
+        """
+        Export one directory if it is not in exported list.
+
+        :return: Export nfs file system succeed or not
+        """
+        if self.is_exported():
+            if self.need_reexport():
+                self.unexport()
+            else:
+                self.already_exported = True
+                logging.warn("Already exported target."
+                             " Don't need export it again")
+                return True
+        export_cmd = "exportfs"
+        if self.options:
+            export_cmd += " -o %s" % ",".join(self.options)
+        export_cmd += " %s:%s" % (self.client, self.path)
+        try:
+            process.system(export_cmd)
+        except process.CmdError, export_failed_err:
+            logging.error("Can not export target: %s" % export_failed_err)
+            return False
+        return True
+
+
+class Nfs(object):
+
+    """
+    Nfs class for handle nfs mount and umount. If a local nfs service is
+    required, it will configure a local nfs server accroding the params.
+    """
+
+    def __init__(self, params):
+        self.mount_dir = params.get("nfs_mount_dir")
+        self.mount_options = params.get("nfs_mount_options")
+        self.mount_src = params.get("nfs_mount_src")
+        self.nfs_setup = False
+        path.find_command("mount")
+        self.mk_mount_dir = False
+        self.unexportfs_in_clean = False
+
+        if params.get("setup_local_nfs") == "yes":
+            self.nfs_setup = True
+            path.find_command("service")
+            path.find_command("exportfs")
+            self.nfs_service = service.SpecificServiceManager("nfs")
+
+            self.export_dir = (params.get("export_dir") or
+                               self.mount_src.split(":")[-1])
+            self.export_ip = params.get("export_ip", "*")
+            self.export_options = params.get("export_options", "").strip()
+            self.exportfs = Exportfs(self.export_dir, self.export_ip,
+                                     self.export_options)
+            self.mount_src = "127.0.0.1:%s" % self.export_dir
+
+    def is_mounted(self):
+        """
+        Check the NFS is mounted or not.
+
+        :return: If the src is mounted as expect
+        :rtype: Boolean
+        """
+        return utils_misc.is_mounted(self.mount_src, self.mount_dir, "nfs")
```

Comment:
This is the first extra dependency. At this stage I don't think you should include the full `utils_misc` which contains tons of generic utils, but you should only include the minimal part of it. And in this case it'd actually be better to create a new library called `mount` to handle mount/unmount/is_mounted/... which would only be dependent on `process`.